File: ClassEval_T/java/solutuon/ExcelProcessor.java
package org.example;

import org.apache.poi.ss.usermodel.Cell;
import org.apache.poi.ss.usermodel.Row;
import org.apache.poi.ss.usermodel.Sheet;
import org.apache.poi.ss.usermodel.Workbook;
import org.apache.poi.xssf.usermodel.XSSFWorkbook;

import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

public class ExcelProcessor {

    public ExcelProcessor() {
    }

    public List<List<Object>> readExcel(String fileName) {
        List<List<Object>> data = new ArrayList<>();
        try (FileInputStream file = new FileInputStream(fileName);
             Workbook workbook = new XSSFWorkbook(file)) {
            Sheet sheet = workbook.getSheetAt(0);
            for (Row row : sheet) {
                List<Object> rowData = new ArrayList<>();
                for (Cell cell : row) {
                    switch (cell.getCellType()) {
                        case STRING:
                            rowData.add(cell.getStringCellValue());
                            break;
                        case NUMERIC:
                            rowData.add((int) cell.getNumericCellValue());
                            break;
                        default:
                            rowData.add(null);
                            break;
                    }
                }
                data.add(rowData);
            }
        } catch (IOException e) {
            return null;
        }
        return data;
    }



    public boolean writeExcel(List<List<Object>> data, String fileName) {
        try (Workbook workbook = new XSSFWorkbook();
             FileOutputStream fileOut = new FileOutputStream(fileName)) {
            Sheet sheet = workbook.createSheet("Sheet1");
            for (int i = 0; i < data.size(); i++) {
                Row row = sheet.createRow(i);
                List<Object> rowData = data.get(i);
                for (int j = 0; j < rowData.size(); j++) {
                    Cell cell = row.createCell(j);
                    Object value = rowData.get(j);
                    if (value instanceof String) {
                        cell.setCellValue((String) value);
                    }
                    else if (value instanceof Integer) {
                        cell.setCellValue((Integer) value);
                    }
                }
            }
            workbook.write(fileOut);
            return true;
        } catch (IOException e) {
            return false;
        }
    }

    public String processExcelData(int N, String saveFileName) {
        List<List<Object>> data = readExcel(saveFileName);
        if (data == null || N >= data.get(0).size()) {
            return null;
        }
        List<List<Object>> newData = new ArrayList<>();
        for (List<Object> row : data) {
            List<Object> newRow = new ArrayList<>(row);
            Object value = row.get(N);
            if (value instanceof String) {
                newRow.add(((String) value).toUpperCase());
            }
            else {
                newRow.add(value);
            }
            newData.add(newRow);
        }
        String newFileName = saveFileName.replace(".xlsx", "_process.xlsx");
        boolean success = writeExcel(newData, newFileName);
        return success ? newFileName : null;
    }
}

File: ClassEval_T/cpp/solution/ExcelProcessor.cpp

#include <OpenXLSX.hpp>
#include <iostream>
#include <variant>
#include <string>
#include <vector>

#include <string>
#include <vector>
#include <variant>

class ExcelProcessor {
public:
    ExcelProcessor() = default;

    std::vector<std::vector<std::variant<std::string, int>>> read_excel(const std::string& file_name);

    int write_excel(const std::vector<std::vector<std::variant<std::string, int>>>& data, const std::string& file_name);

    std::pair<int, std::string> process_excel_data(size_t N, const std::string& save_file_name);
};

std::vector<std::vector<std::variant<std::string, int>>> ExcelProcessor::read_excel(const std::string& file_name) {
    std::vector<std::vector<std::variant<std::string, int>>> data;
    try {
        OpenXLSX::XLDocument doc;
        doc.open(file_name);
        auto sheet = doc.workbook().worksheet("Sheet1");

        unsigned int rowCount = sheet.rowCount();
        unsigned int colCount = sheet.columnCount();

        for (unsigned int rowIndex = 1; rowIndex <= rowCount; ++rowIndex) {
            std::vector<std::variant<std::string, int>> rowData;
            for (unsigned int colIndex = 1; colIndex <= colCount; ++colIndex) {
                auto cell = sheet.cell(rowIndex, colIndex);

                if (cell.value().type() == OpenXLSX::XLValueType::Integer) {
                    rowData.emplace_back(static_cast<int>(cell.value().get<int64_t>()));
                }
                else if (cell.value().type() == OpenXLSX::XLValueType::String) {
                    rowData.emplace_back(cell.value().get<std::string>());
                }
            }
            data.push_back(rowData);
        }
        doc.close();
    }
    catch (...) {
        return {};
    }
    return data;
}

int ExcelProcessor::write_excel(const std::vector<std::vector<std::variant<std::string, int>>>& data, const std::string& file_name) {
    try {
        OpenXLSX::XLDocument doc;
        doc.create(file_name);
        auto sheet = doc.workbook().worksheet("Sheet1");

        for (unsigned int rowIndex = 0; rowIndex < data.size(); ++rowIndex) {
            for (unsigned int colIndex = 0; colIndex < data[rowIndex].size(); ++colIndex) {
                auto cell = sheet.cell(rowIndex + 1, colIndex + 1);
                if (std::holds_alternative<int>(data[rowIndex][colIndex])) {
                    cell.value() = std::get<int>(data[rowIndex][colIndex]);
                }
                else {
                    cell.value() = std::get<std::string>(data[rowIndex][colIndex]);
                }
            }
        }
        doc.save();
        doc.close();
        return 1;
    }
    catch (...) {
        return 0;
    }
}

std::pair<int, std::string> ExcelProcessor::process_excel_data(size_t N, const std::string& save_file_name) {
    auto data = read_excel(save_file_name);
    if (data.empty() || N >= data[0].size()) {
        return { 0, "" };
    }

    std::vector<std::vector<std::variant<std::string, int>>> new_data;
    for (const auto& row : data) {
        std::vector<std::variant<std::string, int>> new_row(row);
        if (auto value = std::get_if<std::string>(&row[N])) {
            if (!std::isdigit((*value)[0])) {
                std::string now = *value;
                for (int i = 0; i < now.size(); ++i)
                {
                    if (now[i] > 'Z')
                    {
                        now[i] -= 32;
                    }
                }
                new_row.emplace_back(now);
            }
            else {
                new_row.emplace_back(value->substr(0, 1));
            }
        }
        else {
            new_row.emplace_back(std::get<int>(row[N]));
        }
        new_data.push_back(new_row);
    }

    std::string new_file_name = save_file_name.substr(0, save_file_name.find_last_of('.')) + "_process.xlsx";
    int success = write_excel(new_data, new_file_name);
    return { success, new_file_name };
}
